src/main/java/uk/gov/companieshouse/api/accounts/validation/IntangibleAssetsValidator.java

```diff
@@ -0,0 +1,286 @@
+
+    private static final String INTANGIBLE_NOTE = "$.intangible_assets";
+    private static final String COST_AT_PERIOD_START = ".cost.at_period_start";
+    private static final String ADDITIONS = ".cost.additions";
+    private static final String DISPOSALS = ".cost.disposals";
+    private static final String REVALUATIONS = ".cost.revaluations";
+    private static final String TRANSFERS = ".cost.transfers";
+    private static final String COST_AT_PERIOD_END = ".cost.at_period_end";
+    private static final String NET_BOOK_VALUE_CURRENT_PERIOD = ".net_book_value_at_end_of_current_period";
+
+    public Errors validateIntangibleAssets(IntangibleAssets intangibleAssets, Transaction transaction, String companyAccountsId, HttpServletRequest request)
+    throws DataException {
+        Errors errors = new Errors();
+
+        try {
+            boolean isMultipleYearFiler = companyService.isMultipleYearFiler(transaction);
+
+            List<IntangibleSubResource> invalidSubResources = new ArrayList<>();
+            verifySubResourcesAreValid(intangibleAssets, errors, isMultipleYearFiler, invalidSubResources);
+            verifyNoteNotEmpty(intangibleAssets, errors, isMultipleYearFiler);
+            validateSubResourceTotals(intangibleAssets, errors, isMultipleYearFiler, invalidSubResources);
+            if(errors.hasErrors()) {
+                return errors;
+            }
+
+        } catch(ServiceException se) {
+            throw new DataException(se.getMessage(), se);
+        }
+
+        return errors;
+    }
+
+    private void verifyNoteNotEmpty(IntangibleAssets intangibleAssets, Errors errors, boolean isMultipleYearFiler) {
+
+        if(intangibleAssets.getGoodwill() == null &&
+                intangibleAssets.getOtherIntangibleAssets() == null &&
+                intangibleAssets.getTotal() == null) {
+
+            addError(errors, valueRequired, getJsonPath(IntangibleSubResource.TOTAL, NET_BOOK_VALUE_CURRENT_PERIOD));
+        }
+    }
+
+    private void verifySubResourcesAreValid(IntangibleAssets intangibleAssets, Errors errors, boolean isMultipleYearFiler, List<IntangibleSubResource> invalidSubResources) {
+
+        validateSubResource(intangibleAssets.getGoodwill(), errors,  isMultipleYearFiler, IntangibleSubResource.GOODWILL, invalidSubResources);
+        validateSubResource(intangibleAssets.getOtherIntangibleAssets(), errors, isMultipleYearFiler, IntangibleSubResource.OTHER_INTANGIBLE_ASSETS, invalidSubResources);
+        validateSubResource(intangibleAssets.getTotal(), errors, isMultipleYearFiler, IntangibleSubResource.TOTAL, invalidSubResources);
+
+    }
+
+    private void validateSubResource(IntangibleAssetsResource intangibleAssetsResource, Errors errors, boolean isMultipleYearFiler, IntangibleSubResource intangibleSubResource, List<IntangibleSubResource> invalidSubResource ) {
+
+        if(intangibleAssetsResource != null) {
+
+            if(isMultipleYearFiler) {
+                //TODO
```

Comment:
Typically we reference a story number in a TODO, for traceability

Reply:
Added a Jira story as suggested